subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -119,6 +119,75 @@ static inline void set_pkt_ll_addr(struct net_linkaddr *addr, bool comp,
 	addr->type = NET_LINK_IEEE802154;
 }
 
+/**
+ * Filters the destination address of the frame (used when IEEE802154_HW_FILTER
+ * is not available).
+ */
+static bool ieeee802154_check_dst_addr(struct net_if *iface,
+				      struct ieee802154_mhr *mhr)
+{
+	struct ieee802154_context *ctx = net_if_l2_data(iface);
+	const struct ieee802154_radio_api *radio =
+		net_if_get_device(iface)->api;
+	uint8_t i;
+
+	/*
+	 * FIXME: should I remove this check so the destination address is
+	 * always filtered? */
+	if (radio && (radio->get_capabilities(net_if_get_device(iface)) &
+		      IEEE802154_HW_FILTER)) {
+		return true;
+	}
+
+	/*
+	 * Apply filtering requirements from chapter 6.7.2 of the 802.15.4 standard:
```

Comment:
```suggestion
	 * Apply filtering requirements from chapter 6.7.2 of the IEEE 802.15.4-2015 standard:
```
The filtering procedure is described in different chapters in different revisions of the specification. Please be precise which revision have you used.